woocommerce/woocommerce-square
# [Cash App Pay] Add Support for buying a gift card product type using Cash App Pay.

### All Submissions:



---

### Changes proposed in this Pull Request:
PR adds support for buying a gift card product type using Cash App Pay. PR also fixes #74 to ensure the gift card is created when the credit card transaction type is "Authorization," and "Charge Virtual-Only Orders" is enabled 

Closes #73 
Closes #74 

### Steps to test the changes in this Pull Request:
#### Test #73
1. Create a gift card product.
1. Go to the product page on the front-end of the web store, choose "Buy a new gift card," enter details, and add the product to the cart.
1. Go to the checkout and verify that Cash App Pay is available in the listed payment methods.
1. Place an order using the Cash App Pay gateway and verify that the order is placed successfully.
1. Verify that the gift card is created and activated.
1. Verify that an email is sent to the recipient.
1. Go to the gift card product page again, choose "Add value to an existing gift card," and add the product to the cart.
Place an order using the Cash App Pay gateway and verify that the order is placed successfully.

#### Test #74
Detailed issue reproduce steps are given in the issue description. follow that and verify that the issue no longer exists.

### Changelog entry
> Add - Support for buying a Square Gift Card using Cash App Pay.
> Fix - Ensure a Gift Card is created when the Credit Card transaction type is "Authorization," and "Charge Virtual-Only Orders" is enabled.
